src/re.rs

```diff
@@ -1079,16 +1076,12 @@ impl<'r, 't> Iterator for FindCaptures<'r, 't> {
 
         // Don't accept empty matches immediately following a match.
         // i.e., no infinite loops please.
-        if e == s && Some(self.last_end) == self.last_match {
-            if self.last_end >= self.search.len() {
-                return None;
-            }
+        if e == s {
```

Comment:
I think the issue here is if a second empty match occurs at the end of the string, then this branch is taken, `self.last_end` is incremented by `1` and ` Some(...)` value is returned. In contrast, the code currently returns a `None`. This error appears to be triggered by the `searcher_many_zero_length_matches` test, where your patch produces a new `Match(6, 6)` that shouldn't be there.